pytorch_lightning/callbacks/progress/progress.py

```diff
@@ -11,329 +11,7 @@
-            self._update_bar(self.test_progress_bar)
-
-    def on_test_end(self, trainer, pl_module):
-        super().on_test_end(trainer, pl_module)
-        self.test_progress_bar.close()
-
-    def on_predict_epoch_start(self, trainer, pl_module):
-        super().on_predict_epoch_start(trainer, pl_module)
-        self.predict_progress_bar = self.init_predict_tqdm()
-        self.predict_progress_bar.total = convert_inf(self.total_predict_batches)
-
-    def on_predict_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
-        super().on_predict_batch_end(trainer, pl_module, outputs, batch, batch_idx, dataloader_idx)
-        if self._should_update(self.predict_batch_idx, self.total_predict_batches):
-            self._update_bar(self.predict_progress_bar)
-
-    def on_predict_end(self, trainer, pl_module):
-        self.predict_progress_bar.close()
-
-    def print(
-        self, *args, sep: str = " ", end: str = os.linesep, file: Optional[io.TextIOBase] = None, nolock: bool = False
-    ):
-        active_progress_bar = None
-
-        if self.main_progress_bar is not None and not self.main_progress_bar.disable:
-            active_progress_bar = self.main_progress_bar
-        elif self.val_progress_bar is not None and not self.val_progress_bar.disable:
-            active_progress_bar = self.val_progress_bar
-        elif self.test_progress_bar is not None and not self.test_progress_bar.disable:
-            active_progress_bar = self.test_progress_bar
-        elif self.predict_progress_bar is not None and not self.predict_progress_bar.disable:
-            active_progress_bar = self.predict_progress_bar
-
-        if active_progress_bar is not None:
-            s = sep.join(map(str, args))
-            active_progress_bar.write(s, end=end, file=file, nolock=nolock)
-
-    def _should_update(self, current, total) -> bool:
-        return self.is_enabled and (current % self.refresh_rate == 0 or current == total)
-
-    def _update_bar(self, bar: Optional[tqdm]) -> None:
-        """Updates the bar by the refresh rate without overshooting."""
-        if bar is None:
-            return
-        if bar.total is not None:
-            delta = min(self.refresh_rate, bar.total - bar.n)
-        else:
-            # infinite / unknown size
-            delta = self.refresh_rate
-        if delta > 0:
-            bar.update(delta)
-
-
-def convert_inf(x: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
-    """The tqdm doesn't support inf/nan values. We have to convert it to None."""
-    if x is None or math.isinf(x) or math.isnan(x):
-        return None
-    return x
-
-
-def reset(bar: tqdm, total: Optional[int] = None) -> None:
-    """Resets the tqdm bar to 0 progress with a new total, unless it is disabled."""
-    if not bar.disable:
-        bar.reset(total=convert_inf(total))
+# points to the default implementation
+ProgressBar = TQDMProgressBar
```

Comment:
rather inherit and add a deprecation warning, see what we did with Metrics

Reply:
for example #6584

Reply:
Like this

```py
class ProgressBar(TQDMProgressBar):
    pass
```